docs(livemode): add livemode documentation

MichalHe opened the pull request:
This commit introduces a high-level documentation of the livemode
feature. The following is documented:
 - how the upgrade process differs from the standard one when using
   livemode
 - why is the feature useful, i.e., why would anyone use it
 - how to enable and use the feature
 - what configuration options are available, including an example of
   configuration

Requires #1350

jira-ref: RHELMISC-9703

pirat89 commented on docs/source/livemode.md:
```diff
@@ -0,0 +1,87 @@
+# Livemode - having a more familiar, possibly interactive, upgrade environment
+**Note that the livemode feature is experimental.**
```
Note that the suggestion is littlebit malformed and not sure how to correctly input triple backticks here :) but you get the idea I believe.

```suggestion
```{warning}
Note that the livemode feature is experimental.
\`\`\`
```

MichalHe replied:
Suggestion applied to the `index.rst` of `experimental-features` subsection.

pirat89 commented on docs/source/livemode.md:
```diff
@@ -0,0 +1,87 @@
+# Livemode - having a more familiar, possibly interactive, upgrade environment
+**Note that the livemode feature is experimental.**
+
+Leapp features an experimental feature named _livemode_ that partially replaces
```
The sentense is kind of hard to get, let's simplify it to something like

```suggestion
_livemode_ is an experimental feature that partially replaces
```

MichalHe replied:
Suggestion applied :+1:

pirat89 commented:
@MichalHe I think we should create a dedicated section for experimental features. We should not put it into docs just like that but rather having a dedicated section for it. I am thinking about putting it under:
```
 * Configuring in-place upgrade
    * ... 
    * Experimental Features
      * LiveMode
```

When we move it in future from the section experimental features somewhere else, we can set a rule for relocation from the org URL to the new one. I see now that configuration section is not "separated" correctly, so it would need small adjustment as well. wdyt?

Commit:
docs(configuring-ipu): add experimental features section

With the upcoming introduction of documentation of the LiveMode feature,
a question arises where to put the documentation. This patch introduces
a new subsection 'Experimental features' under the 'Configuring IPU'
section, so we have one place to where to put such features.

Commit:
docs(livemode): add LiveMode documentation

This commit introduces a high-level documentation of the livemode
feature. The following is documented:
 - how the upgrade process differs from the standard one when using
   livemode
 - why is the feature useful, i.e., why would anyone use it
 - how to enable and use the feature
 - what configuration options are available, including an example of
   configuration

Jira-ref: RHELMISC-9703

MichalHe commented:
@pirat89 Done, I have restructured the documentation to include a section named _Experimental features_ under the _Configuring IPU_ section as you have suggested. I have also adopted the "camel case" naming of the feature in the documentation _LiveMode_.

pirat89 reviewed (approved):
nice! merging
